public/app/features/datasources/NewDataSourcePage.tsx

```diff
@@ -32,47 +58,166 @@ class NewDataSourcePage extends PureComponent<Props> {
     this.props.setDataSourceTypeSearchQuery(value);
   };
 
+  renderTypes(types: DataSourcePluginMeta[]) {
+    if (!types) {
+      return null;
+    }
+
+    // apply custom sort ranking
+    types.sort((a, b) => {
+      const aSort = this.sortingRules[a.id] || 0;
+      const bSort = this.sortingRules[b.id] || 0;
+      if (aSort > bSort) {
+        return -1;
+      }
+      if (aSort < bSort) {
+        return 1;
+      }
+
+      return a.name > b.name ? -1 : 1;
+    });
+
+    return (
+      <div className="add-data-source-grid">
+        {types.map((plugin, index) => (
+          <DataSourceTypeCard
+            key={`${plugin.id}-${index}`}
+            plugin={plugin}
+            onClick={() => this.onDataSourceTypeClicked(plugin)}
+            onLearnMoreClick={this.onLearnMoreClick}
+          />
+        ))}
+      </div>
+    );
+  }
+
+  onLearnMoreClick = (evt: React.SyntheticEvent<HTMLElement>) => {
+    evt.stopPropagation();
```

Comment:
Why are we stoping the propagation here? To prevent event capture on the card itself?

Reply:
the whole card is clickable/selectable just want the Learn more to not trigger the card select

Reply:
cool, that was my guess